bootspec/src/generation.rs

```diff
@@ -33,6 +33,18 @@ impl Generation {
     }
 }
 
+impl TryFrom<Generation> for v1::GenerationV1 {
+    type Error = crate::BootspecError;
+
+    fn try_from(value: Generation) -> Result<Self, Self::Error> {
```

Comment:
While this current implementation is infallible (because we don't have any other variants), it will be come fallible as soon as we add a new variant.